### PR TITLE
made scale button and attached scale query logic

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: public/audioplayer.html [hunk 1]
```diff
@@ -65,4 +71,13 @@
         })
     }
 
+    function playScale() {
+        let context = new (window.AudioContext || window.webkitAudioContext)();
+        let note = new Sound(context);
+        let now = context.currentTime;
+        queryObject.scaleNotes.forEach(array => {
+            note.play(array[0], now + array[1], now + array[2]);
+        })
```
Comment (dancingshell) [inline public/audioplayer.html]: you pull pull all this into a function that takes the notes to play as a param
```js
function playNotes(notes) {
   let context = new (window.AudioContext || window.webkitAudioContext)();
   let note = new Sound(context);
   let now = context.currentTime;
   notes.forEach(array => {   
     note.play(array[0], now + array[1], now + array[2]);
   })
}
```

and then each specific function can know how to get the right notes and pass it along
```js
function playScale() {
  playNotes(queryObject.scaleNotes)
}

function playMelody() {
  playNotes(queryObject.melody)
}
```

File: src/melodyMaker.js [hunk 2]
```diff
@@ -20,13 +33,8 @@ function createMelody() {
...
         time += thisNoteLength
     }
-    validNotes.forEach((note) => {
-        notesArray.push([noteFrequencyMap[note][2], time, time + 1])
-        time += 1
-    })
 
-    notesArray.push([noteFrequencyMap[validNotes[0]][2], time, time + 3])
-    var melodyParameters = [notesArray, tonic[0], scaleNames]
+    var melodyParameters = [notesArray, tonic[0], scaleNames, createScale(tonic, scale)]
```
Comment (dancingshell) [inline src/melodyMaker.js:38]: if you return an object here you dont have to know what index to lookup on the other side. instead you can look up by key name

**Review by dancingshell — APPROVED** [review]

File: public/audioplayer.html (hunk 1 above)
Comment (ghost) [reply public/audioplayer.html]: yess thanks!

File: src/melodyMaker.js (hunk 2 above)
Comment (ghost) [reply src/melodyMaker.js:38]: great call, I'll get this in a fast-follow

**Pull request merged**